Structure.Rmd

```diff
@@ -0,0 +1,398 @@
+t
+
+## Naming your package {#naming}
+
+> "There are only two hard things in Computer Science: cache invalidation and 
+> naming things." 
+>
+> --- Phil Karlton
+
+Before you can create your first package, you need to come up with a name for it. I think this is the hardest part of creating a package! (Not least because devtools can't automate it for you.)
+
+### Requirements for a name
+
+There are three formal requirements: the name can only consist of letters, numbers and periods, i.e., `.`; it must start with a letter; and it cannot end with a period. Unfortunately, this means you can't use either hyphens or underscores, i.e., `-` or `_`, in your package name. I recommend against using periods in package names because it has confusing connotations (i.e., file extension or S3 method).
+
+### Strategies for creating a name
+
+If you're planning on releasing your package, I think it's worth spending a few minutes to come up with a good name. Here are some recommendations for how to go about it:
+
+* Pick a unique name you can easily Google. This makes it easy for potential users to 
+  find your package (and associated resources) and for you to see who's using it. 
+  You can also check if a name is already used on CRAN by loading <http://cran.r-project.org/web/packages/[PACKAGE_NAME]>.
+
+* Avoid using both upper and lower case letters: doing so makes the package name
+  hard to type and even harder to remember. For example, I can never remember if it's
+  Rgtk2 or RGTK2 or RGtk2.
+
+* Find a word that evokes the problem and modify it so that it's unique:
+
+    * plyr is generalisation of the apply family, and evokes pliers.
+    * lubridate makes dates and times easier.
+    * knitr (knit + r) is "neater" than sweave (s + weave).
+    * testdat tests that data has the correct format.
+
+* Use abbreviations: 
+
+    * Rcpp = R + C++ (plus plus)
+    * lvplot = letter value plots. 
+
+* Add an extra R: 
+
+    * stringr provides string tools.
+    * tourr implements grand tours (a visualisation method).
+    * gistr lets you programmatically create and modify GitHub gists.
+
+If you're creating a package that talks to a commercial service, make sure you check the branding guidelines to avoid problems down the line. For example, rDrop isn't called rDropbox because Dropbox prohibits any applications from using the full trademarked name.
```

Comment:
Need to mention available here (with caution about interpreting urban dictionary results)